torch/csrc/jit/codegen/cuda/fusion.h

```diff
@@ -191,6 +197,12 @@ class TORCH_CUDA_CU_API Fusion final {
   bool hasInput(const Val* val) const;
   bool hasOutput(const Val* val) const;
 
+ protected:
+  friend SegmentCandidateFinder;
+  friend SegmentedFusion;
+
+  static IrCloner copy(const Fusion* from, Fusion* to);
```

Comment:
Why do we need this? 

This makes me a bit nervous - it seems we're trying to use the copy operation in an .. "unexpected" way.

Reply:
I was slightly unsure about this too. The returned `IrCloner` was used as a map from the original fusion to the copied fusion so it's easier to navigate in the copied fusion. e.g:
[https://github.com/csarofeen/pytorch/blob/multi_kernel_merged/torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp#L466-L482](https://github.com/csarofeen/pytorch/blob/multi_kernel_merged/torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp#L466-L482)
 This use is only limited to the friend classes. Refactor and optimization are definitely possible but I'm not sure if there is anything we can do quickly under the current state.

Reply:
The same topic has been discussed in https://github.com/csarofeen/pytorch/pull/555

What we need is name to node lookup, then the solution is clean and simple:

```
  fusion0.removeOutput(fusion0.lookupNode(tv3->name()));
  fusion0.addOutput(fusion0.lookupNode(tv1->name()));
```